src/components/inputs/SearchSelect.jsx

```diff
@@ -0,0 +1,427 @@
+import React, { PropTypes, addons } from 'react/addons';
+const { update } = addons;
+import _ from 'lodash';
+import classNames from 'classnames';
+
+import InputMixin from '../../mixins/InputMixin';
+
+
+const KEY_CODES = {
+  TAB: 9,
+  ENTER: 13,
+  ARROW_UP: 38,
+  ARROW_DOWN: 40,
+};
+const DEFAULT_PLACEHOLDER = 'Search option';
+
+const optionPropType = PropTypes.shape({
+  id: PropTypes.string.isRequired,
+  label: PropTypes.string.isRequired,
+});
+
+const optionGroupOf = (_optionPropType) => PropTypes.oneOfType([
+  PropTypes.shape({
+    isGroup: PropTypes.bool.isRequired,
+    id: PropTypes.string.isRequired,
+    label: PropTypes.string.isRequired,
+    options: PropTypes.arrayOf(_optionPropType).isRequired,
+  }),
+  optionPropType,
+]);
+
+const OptionDefault = React.createClass({
+
+  displayName: 'Option',
+
+  propTypes: {
+    option: PropTypes.shape({
+      ...optionPropType,
+      label: PropTypes.string.isRequired,
```

Comment:
You can't spread a PropType, as it is a function. `label` is already present in optionPropType anyway.